### PR TITLE
add minetest.get_last_run_mod to lua_api.txt

### PR DESCRIPTION
**Pull request by HybridDog**

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -2274,6 +2274,10 @@ These functions return the leftover itemstack.
     * Call the function `func` after `time` seconds, may be fractional
     * Optional: Variable number of arguments that are passed to `func`
 
+### Last run mod
+* `minetest.get_last_run_mod()`
+    * Returns the name of the last run mod or "*builtin*"
```
Comment (HybridDog) [inline doc/lua_api.txt]: l think sometimes it's even "???"

**Comment by t4im** [comment]
What for? Iirc the function was a workaround for getting the current modname when certain errors happened and only seems directly useful from within builtin, and perhaps not even there anymore at some point.

What could you use it possibly for, that `get_current_modname` doesn't provide you from a mod?

**Comment by HybridDog** [comment]
What could you use it for[…]?

My [function_delayer](https://github.com/HybridDog/function_delayer) mod uses it.
It's useful to some mods' APIs.
Sorry, l don't understand the rest of the question.

**Comment by HybridDog** [comment]
Why don't you merge this?

**Comment by paramat** [comment]
This should not be in it's own section, it should be in `### Misc.` or what ever section is most suitable.

**Comment by HybridDog** [comment]
Ok, fixed.

**Comment by paramat** [comment]
:+1:

**Comment by paramat** [comment]
Some are suggesting this should not be documented snce it is internal.

**Comment by rubenwardy** [comment]
What are the use cases? This is quite unreliable. If it's to identify the mod, there are better ways (eg: by getting the mod to get an object at load time and then use that object to do more stuff). That way, if a mod (=get_last_run_mod) calls code of the another mod (eg, its API or via callbacks) it will use the correct API

**Comment by HybridDog** [comment]
It's not always sufficient to have the api know the mod name at load time.
https://github.com/HybridDog/function_delayer/blob/master/init.lua#L34
Sorry, l don't understand your example.

**Pull request closed without merging**